src/components/app/ProfileLoadingMessage.js

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import explicitConnect from '../../utils/connect';
+
+import { getProfileLoadingState } from '../../selectors/profile';
+
+import type { ProfileLoadingStep } from 'firefox-profiler/types';
+
+import type { ConnectedProps } from '../../utils/connect';
+
+type StateProps = {|
+  +profileLoadingStep: ProfileLoadingStep,
+  +progress: number,
+|};
+
+type Props = ConnectedProps<{||}, StateProps, {||}>;
+
+class ProfileLoadingMessage extends React.PureComponent<Props> {
+  // updateProfileLoadingMessage = () => {};
+
+  // componentDidMount() {
+  //   this.updateProfileLoadingMessage();
+  // }
+
+  // componentDidUpdate() {
+  //   this.updateProfileLoadingMessage();
+  // }
+
+  render() {
+    const { profileLoadingStep, progress } = this.props;
+    return (
+      <div className="profileLoadingMessageContainer">
+        <div className="loadingStep">Current Step: {profileLoadingStep}</div>
+        <div className="loadingStepProgress">Progress: {progress}</div>
```

Comment:
This will probably need some design consideration before landing, but this is a good way to get feedback on progress.

One other thing to note, is that we often completely block the main thread of the content process while processing the profile. This means JS won't run for updates. This is something we might have to take care to do correctly when updating these components.

I'm also not seeing the progress being used yet, but perhaps we have the necessary information to update it on network requests, but we definitely won't be able to do it on profile processing.